viewing_party/party.py

```diff
@@ -0,0 +1,140 @@
+def create_movie(movie_title, genre, rating):
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    for key in new_movie:
+        if new_movie[key] == None:
+            return None
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    total_rating = 0.0
+    
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"] 
+    
+    if len(user_data["watched"]) > 0:
+        avg_rating = total_rating/len(user_data["watched"])
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_dict = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_dict:
+            genre_dict[genre] += 1
+        else:
+            genre_dict[genre] = 1
+
+    most_freq = genre
+    for genre, freq in genre_dict.items():
+        if freq > genre_dict[most_freq]:
+            most_freq = genre
+
+    return most_freq
+
+
+def get_unique_watched(user_data):
+
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie)
+    
+    friends_watched = []
+    for watched_dict in user_data["friends"]:
+        for title in watched_dict["watched"]:
+            friends_watched.append(title)
```

Comment:
Notice the items we're iterating over here are full movie dictionaries, not titles. So `movie` might be a better iteration variable name, or `friend_movie` if you wanted to help keep it distinct from the previously used `movie` during the user loop.